File: 18. Remove duplicates sorted.py
class Node:
    def __init__(self,data): self.data=data; self.next=None

def remove_dup(head):
    cur=head
    while cur and cur.next:
        if cur.data==cur.next.data:
            cur.next=cur.next.next
        else:
            cur=cur.next
    return head

def print_list(h):
    cur=h; out=[]
    while cur:
        out.append(str(cur.data)); cur=cur.next
    print(' -> '.join(out))

if __name__=='__main__':
    head=Node(1); head.next=Node(1); head.next.next=Node(2); head.next.next.next=Node(3); head.next.next.next.next=Node(3)
    head=remove_dup(head)
    print_list(head)

File: 18. Remove duplicates sorted.cpp
#include <iostream>
using namespace std; struct Node{int data; Node* next; Node(int d):data(d),next(nullptr){}
};
void removeDup(Node* head){ if(!head) return; Node* cur=head; while(cur->next){ if(cur->data==cur->next->data){ Node* t=cur->next; cur->next=t->next; delete t; } else cur=cur->next; } }
void print(Node* h){ while(h){ cout<<h->data; if(h->next) cout<<" -> "; h=h->next;} cout<<endl; }
int main(){ Node* head=new Node(1); head->next=new Node(1); head->next->next=new Node(2); head->next->next->next=new Node(3); head->next->next->next->next=new Node(3); removeDup(head); print(head); return 0; }
